processing-chain/src/main/groovy/org/orbisgis/processingchain/CreateScalesOfAnalysis.groovy

```diff
@@ -0,0 +1,64 @@
+package org.orbisgis.processingchain
+
+import groovy.transform.BaseScript
+import org.orbisgis.SpatialUnits
+import org.orbisgis.processmanager.ProcessMapper
+
+@BaseScript ProcessingChain processingChain
+
+/** The processing chain creates the units used to describe the territory at three scales: Reference Spatial
+ * Unit (RSU), block and building. The creation of the RSU needs several layers such as the hydrology,
+ * the vegetation, the roads and the rail network and the boundary of the study zone. The blocks are created
+ * from the buildings that are in contact.
+ * Then the relationships between each scale is initialized in each unit table: the RSU ID is stored in
+ * the block and in the building tables whereas the block ID is stored only in the building table.
+ *
+ * @param zoneTable The area of zone to be processed
+ * @param roadTable The road table to be processed
+ * @param railTable The rail table to be processed
+ * @param vegetationTable The vegetation table to be processed
+ * @param hydrographicTable The hydrographic table to be processed
+ * @param surface_vegetation The minimum area of vegetation that will be considered to delineate the RSU (default 100,000 m²)
+ * @param surface_hydro  The minimum area of water that will be considered to delineate the RSU (default 2,500 m²)
+ * @param inputTableName The input table where are stored the geometries used to create the block (e.g. buildings...)
+ * @param distance A distance to group two geometries (e.g. two buildings in a block - default 0.01 m)
```

Comment:
There is any distance parameter in the mapper ?

Reply:
Yes, actually we use this parameter as input of the mapper since it is an input of the "createBlocks" IProcess. If two buildings are very close, they are considered as neighbors and they are merged. But I am actually not sure it is a good idea to set 0.01 m as default value since some of the indicators use the merged buildings created on the fly in their own IProcess. Then it would lead to a difference of geometries used as input between the IProcesses. May be setting to 0 in the "createBlocks" IProcess for now is the right solution and we will see later if we get a problem with geometries.

Reply:
+100